packages/angular/cli/utilities/check-package-manager.ts

```diff
@@ -5,49 +5,54 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
+import { execSync } from 'child_process';
+import { existsSync } from 'fs';
+import { join } from 'path';
+import { getPackageManager as getConfiguredPackageManager } from './config';
 
-import { terminal } from '@angular-devkit/core';
-import { exec } from 'child_process';
-import { promisify } from 'util';
-import { getPackageManager } from './config';
 
-const execPromise = promisify(exec);
-const packageManager = getPackageManager();
+export function supportsYarn(): boolean {
+  try {
+    execSync('yarn --version');
 
+    return true;
+  } catch {
+    return false;
+  }
+}
 
-export function checkYarnOrCNPM() {
+export function supportsNpm(): boolean {
+  try {
+    execSync('npm --version');
 
-  // Don't show messages if user has already changed the default.
-  if (packageManager !== 'default') {
-    return Promise.resolve();
+    return true;
+  } catch {
+    return false;
   }
-
-  return Promise
-      .all([checkYarn(), checkCNPM()])
-      .then((data: Array<boolean>) => {
-        const [isYarnInstalled, isCNPMInstalled] = data;
-        if (isYarnInstalled && isCNPMInstalled) {
-          console.error(terminal.yellow('You can `ng config -g cli.packageManager yarn` '
-            + 'or `ng config -g cli.packageManager cnpm`.'));
-        } else if (isYarnInstalled) {
-          console.error(terminal.yellow('You can `ng config -g cli.packageManager yarn`.'));
-        } else if (isCNPMInstalled) {
-          console.error(terminal.yellow('You can `ng config -g cli.packageManager cnpm`.'));
-        } else  {
-          if (packageManager !== 'default' && packageManager !== 'npm') {
-            console.error(terminal.yellow(`Seems that ${packageManager} is not installed.`));
-            console.error(terminal.yellow('You can `ng config -g cli.packageManager npm`.'));
-          }
-        }
-      });
 }
 
-function checkYarn() {
-  return execPromise('yarn --version')
-    .then(() => true, () => false);
-}
+export function getPackageManager(root: string): string {
```

Comment:
Filename suggests the main export is `checkPackageManager` but instead it is `getPackageManager`. One of these should probably change. From the code, it looks like `getPackageManager` is the right name.

Reply:
Done.